server-spi/src/main/java/org/keycloak/vault/VaultTranscriber.java

```diff
@@ -0,0 +1,112 @@
+     * value string. The value must follow the format {@code ${vault.<KEY>}} where {@code <KEY>} identifies the entry in
+     * the vault. If the value doesn't follow the vault expression format, it is assumed to be the secret itself and is
+     * encoded into a {@link VaultCharSecret}.
+     * <p/>
+     * To avoid unnecessary encoding it is recommended that the {@link #isVaultExpression(String)} method is used to check
+     * if the value is a vault expression and then call this method only when that is true.
+     * <p/>
+     * The returned {@link VaultCharSecret} extends {@link AutoCloseable} and it is strongly recommended that it is used in
+     * try-with-resources blocks to ensure the raw secret is overridden (destroyed) when the calling code is finished using
+     * it.
+     *
+     * @param value a {@link String} that might be a vault expression containing a vault entry key.
+     * @return a {@link VaultRawSecret} representing the secret that was read from the vault. If the specified value is not
+     *         a vault expression then the returned secret is the value itself encoded as a {@link VaultRawSecret}.
+     */
+    VaultCharSecret getCharSecret(final String value);
+
+    /**
+     * Obtains the secret represented as a {@link String} from the vault that matches the entry in the specified value.
+     * The value must follow the format {@code ${vault.<KEY>}} where {@code <KEY>} identifies the entry in the vault. If
+     * the value doesn't follow the vault expression format, it is assumed to be the secret itself.
+     * <p/>
+     * To avoid unnecessary manipulation of the value string it is recommended that the {@link #isVaultExpression(String)}
+     * method is used to check if the value is a vault entry string and then call this method only when that is true.
+     * <p/>
+     * <b>WARNING:</b> this method, unlike the other variations, doesn't return an {@link AutoCloseable} secret because
+     * strings cannot be overridden (destroyed) so there's no point in having a special type to enclose the secret
+     * string for usage in try-with-resources blocks. This means the secret might be visible in memory for as long as the
+     * string object exists. For this reason, the constructed string is returned as a {@link WeakReference} to convey that
+     * the secret string shouldn't be held longer than needed and to make sure it is available for the GC as soon as possible.
+     *
+     * @param value a {@link String} that might be a vault expression containing a vault entry key.
+     * @return an {@link Optional} holding the weak reference to the constructed secret string. If the specified value is not
+     *         a vault expression then the returned secret is the value itself enclosed in a {@link WeakReference}.
+     */
+    Optional<WeakReference<String>> getStringSecret(final String value);
```

Comment:
This is not inconsistent with the rest methods. To have it consistent, I suggest returning some `VaultStringSecret` object that would be `Closeable` and work with `WeakReference<String>`, where `close()` would call `ref.clear()`.

Reply:
Yeah, it caught my eye when @mhajas tried to write a use case for the transcriber and I wondered why there is no `try` there. Thanks, and sorry for confusion here.

Reply:
Reintroduced the VaultStringSecret class.